### PR TITLE
JavaScript functions and signatures

### PR DESCRIPTION
**Pull request by nilete22**

### CODE REVIEW

File: exercise_01/getDistanceFromThreshold.js [hunk 1]
```diff
@@ -4,3 +4,18 @@
...
+
+/**
+ * Calculate how much a number exceeds or is left to reach a threshold.
+ * @param {number} number
+ * @param {number} threshold
+ * @returns {number}
+ */
+function getDistanceFromThreshold(number, threshold) {
+  if (typeof number !== "number" || typeof threshold !== "number") {
+    throw new Error("Both arguments must be numbers");
```
Comment (tictools) [inline exercise_01/getDistanceFromThreshold.js]: We haven't still learnt how to use `throw`.
Do you know how to properly use it?
Have you tested it in your `index.js`? How are you controlling this error? 🤔

File: exercise_02/getTotalWordsFromString.js [hunk 2]
```diff
@@ -3,3 +3,19 @@
...
  */
+
+/**
+ * Get the total number of words in a string.
+ * @param {string} inputString
+ * @returns {number}
+ */
+function getTotalWordsFromString(inputString) {
+  if (typeof inputString !== "string") {
+    throw new Error("Input must be a string");
```
Comment (tictools) [inline exercise_02/getTotalWordsFromString.js]: Same here regarding `throw`... 🤔

File: exercise_02/index.js [hunk 3]
```diff
@@ -1 +1,5 @@
-//import 'getTotalWordsFromString' and execute
+import getTotalWordsFromString from "./getTotalWordsFromString.js";
+
+const inputString = "Hello World, Im Nil";
+
+console.log(getTotalWordsFromString(inputString));
```
Comment (tictools) [inline exercise_02/index.js:5]: How are you controlling thrown error here if passed argument is not a `string`?

File: exercise_03/index.js [hunk 4]
```diff
@@ -1 +1,5 @@
-//import 'generateRandomNumberInRange' and execute
+import generateRandomNumberInRange from "./generateRandomNumberInRange.js";
+
+const limit = 10;
+
+console.log(generateRandomNumberInRange(limit));
```
Comment (tictools) [inline exercise_03/index.js:5]: Same here: how do yo control error?

File: exercise_04/index.js [hunk 5]
```diff
@@ -1 +1,3 @@
-//import 'isGreaterThan' and execute
+import isGreaterThan from "./isGreaterThan.js";
+
+console.log(isGreaterThan(10, 1));
```
Comment (tictools) [inline exercise_04/index.js:3]: Same here: how do yo control error?

File: exercise_05/index.js [hunk 6]
```diff
@@ -1 +1,3 @@
-//import 'isAdditionGreaterThanFifty' and execute
+import isAdditionGreaterThanFifty from "./isAdditionGreaterThanFifty.js";
+
+console.log(isAdditionGreaterThanFifty(35, 25));
```
Comment (tictools) [inline exercise_05/index.js:3]: Same here: how do yo control error?

File: exercise_06/index.js [hunk 7]
```diff
@@ -1 +1,8 @@
-//import 'checkCollectionHasElements' and execute
+import checkCollectionHasElements from "./checkCollectionHasElements.js";
+
+const collection = [1, 2, 3];
+const emptyCollection = [];
+
+console.log(checkCollectionHasElements(collection));
```
Comment (tictools) [inline exercise_06/index.js:6]: Same here: how do yo control error?

File: exercise_07/index.js [hunk 8]
```diff
@@ -1 +1,11 @@
-//import isTypeOf' and execute
+import isTypeOf from "./isTypeOf.js";
+
+const stringValue = "Hello";
+const numberValue = 12;
+const booleanValue = true;
+const arrayValue = [1, 2, 3];
+
+console.log(isTypeOf(stringValue, "string"));
```
Comment (tictools) [inline exercise_07/index.js:8]: Same here: how do yo control error?

File: exercise_07/isTypeOf.js [hunk 9]
```diff
@@ -3,6 +3,36 @@
...
  */
+
+/**
+ * Check if the type of a value matches the specified type.
+ * @param {string | number | boolean | Array} value - The input value.
+ * @param {string} type - The type to compare against ('string' | 'number' | 'boolean' | 'array').
+ * @returns {boolean} - True if the type matches, false otherwise.
+ */
+
+function isTypeOf(value, type) {
```
Comment (tictools) [inline exercise_07/isTypeOf.js:17]: You can simplify this logic. Avoid using `switch` if you can implement simple `if` structures.

FIX: simplify logic without `switch`

**Review by tictools — CHANGES_REQUESTED** [review]

File: package.json [hunk 10]
```diff
@@ -9,5 +9,8 @@
   },
```
Comment (MartiCobos99) [inline package.json:11]: Lint and Lint:Fix are missing

**Review by MartiCobos99 — CHANGES_REQUESTED** [review]
Fix the package.json file!

File: .github/workflow/pipeline.yml
Comment (tictools) [inline .github/workflow/pipeline.yml:1]: FIX: rename folders. Job is not executed

**Review by tictools — CHANGES_REQUESTED** [review]

**Review by tictools — APPROVED** [review]

**Pull request merged**